pallets/gear/src/tests.rs

```diff
@@ -3393,6 +3393,90 @@ fn program_messages_to_paused_program_skipped() {
     })
 }
 
+#[test]
+fn locking_gas_for_waitlist() {
+    use demo_gas_burned::WASM_BINARY as GAS_BURNED_BINARY;
+    use demo_gasless_wasting::{InputArgs, WASM_BINARY as GASLESS_WASTING_BINARY};
+
+    let wat = r#"
+    (module
+        (import "env" "memory" (memory 1))
+        (import "env" "gr_wait" (func $gr_wait))
+        (export "handle" (func $handle))
+        (func $handle call $gr_wait)
+    )"#;
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        // This program just waits on each handle message.
+        let waiter = upload_program_default(USER_1, ProgramCodeKind::Custom(wat))
+            .expect("submit result was asserted");
+
+        // This program just does some calculations (burns gas) on each handle message.
+        assert_ok!(Gear::upload_program(
+            Origin::signed(USER_1),
+            GAS_BURNED_BINARY.to_vec(),
+            Default::default(),
+            Default::default(),
+            100_000_000_000,
```

Comment:
Default gas limit can't cover calculations done in its init. But I see no reason to do some proper calculations here, because it adds at least 15 lines of code, while we are not interested in it: we need just initialise the program